code/datums/mind.dm

```diff
@@ -769,17 +762,44 @@
 		var/objective_pos
 		var/def_value
 
+
+
+		var/datum/antagonist/target_antag
+
 		if (href_list["obj_edit"])
 			objective = locate(href_list["obj_edit"])
 			if (!objective)
 				return
-			objective_pos = objectives.Find(objective)
+			
+			for(var/datum/antagonist/A in antag_datums)
+				if(objective in A.objectives)
+					target_antag = A
+					objective_pos = A.objectives.Find(objective)
+					break
+			
+			if(!target_antag) //Shouldn't happen
+				stack_trace("objective without antagonist found")
+				objective_pos = objectives.Find(objective)
 
 			//Text strings are easy to manipulate. Revised for simplicity.
 			var/temp_obj_type = "[objective.type]"//Convert path into a text string.
 			def_value = copytext(temp_obj_type, 19)//Convert last part of path into an objective keyword.
 			if(!def_value)//If it's a custom objective, it will be an empty string.
 				def_value = "custom"
+		else
+			switch(antag_datums.len)
+				if(0)
+					target_antag = add_antag_datum(/datum/antagonist/custom)
+				if(1)
+					target_antag = antag_datums[1]
+				else
+					var/target = input("Which antagonist gets the objective:", "Antagonist", def_value) as null|anything in antag_datums + "(new custom antag)"
+					if (QDELETED(target))

```

Comment:
`code/datums/mind.dm:797:error: target.gc_destroyed: undefined var`

Reply:
ugh yeah, you might need to type to var/datum/target